6_visualize/src/prep_spark_line_fun.R

```diff
@@ -54,17 +55,32 @@ prep_spark_funs_data <- function(stage_data, site_data, timestep_ind, spark_conf
       shapes[[site]] <- list(list())
       next
     }
-
-    # look for gaps so we can break lines/polygons into chunks
-    data_chunks_meta <- rle(is.na(stage_data_i$stage_normalized)) %>% {
-      data_frame(
-        is_gap=.$values,
+    # look for gaps and going above/below freezing so we can break lines/polygons into chunks
+    rle_gaps <- rle(is.na(stage_data_i$stage_normalized))
+    rle_freezing <- rle(stage_data_i$temp_f <= 32)
+    #need to only set is_gap to true if
+    data_chunks_meta <- rle_freezing %>% {
+      tibble(
         end=cumsum(.$lengths),
         start=end - .$lengths + 1,
         before=ifelse(start-1==0, NA, start-1),
         after=ifelse(end+1>nrow(stage_data_i), NA, end+1),
+        is_gap=FALSE,
         duration_hr=as.numeric(stage_data_i$dateTime[end] - stage_data_i$dateTime[start], units='hours'))
     }
+    #look at gaps: do they overlap with any below freezing chunks?
+    #if yes, overwrite that part with a gap
+    if(length(rle_gaps$lengths) > 1) {
+      below_freezing_indices <- which(stage_data_i$temp_f <= 32)
+      gap_indices <- which(is.na(stage_data_i$stage_normalized))
+      overlap <- which(below_freezing_indices %in% gap_indices)
+      if(length(overlap > 0)) {
+        #only one gap in this data, at end; hacky solution for now to just create gap at end
+        # this won't generalize to gaps that start and end
+        browser()
```

Comment:
probably good idea to delete the `browser()` call :)

Reply:
I can actually delete this chunk of code, I ended up just truncating the time frame to avoid the issue